underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -91,6 +91,11 @@
   /** AWS-SDK S3 client. */
   private final AmazonS3Client mClient;
 
+  // TODO(binfan): make this a member of ObjectUnderFileSystem rather than a calculated string
+  // each time on getRootKey() for efficiency
```

Comment:
Isn't this already done?

Reply:
Oh I see, this is only for S3A at the moment. It would be nice to put this in the abstract class if this performance improvement is nontrivial.

Reply:
Yes, will do that in a follow up PR to minimize the change of this one